spec/controllers/equipment_models_controller_spec.rb

```diff
@@ -1,421 +1,371 @@
+                               start_date: Time.zone.today - 1.day,
+                               due_date: Time.zone.today + 1.day)
+          end
+
+          it_behaves_like 'calculates availability correctly', 2
+        end
+
+        context 'with upcoming reservation' do
+          before do
+            FactoryGirl.create(:valid_reservation,
+                               equipment_model: model,
+                               start_date: Time.zone.today + 1.day,
+                               due_date: Time.zone.today + 1.day)
+          end
+
+          it_behaves_like 'calculates availability correctly', 1
+        end
+
+        context 'with overdue reservation' do
+          before do
+            FactoryGirl.create(:overdue_reservation,
+                               equipment_model: model,
+                               start_date: Time.zone.today - 1.day,
+                               due_date: Time.zone.today - 1.day,
+                               equipment_item: model.equipment_items.first)
+            # needed to persist overdue_count
+            model.save
+          end
+
+          it_behaves_like 'calculates availability correctly', 1
+        end
+
+        context 'with checked out reservation' do
+          before do
+            FactoryGirl.create(:checked_out_reservation,
+                               equipment_model: model,
+                               start_date: Time.zone.today - 1.day,
+                               due_date: Time.zone.today + 1.day,
+                               equipment_item: model.equipment_items.first)
+          end
+
+          it_behaves_like 'calculates availability correctly', 1
+        end
+
+        context 'with upcoming and overdue reservations' do
+          before do
+            FactoryGirl.create(:valid_reservation,
+                               equipment_model: model,
+                               start_date: Time.zone.today + 1.day,
+                               due_date: Time.zone.today + 1.day)
+            FactoryGirl.create(:overdue_reservation,
+                               equipment_model: model,
+                               start_date: Time.zone.today - 1.day,
+                               due_date: Time.zone.today - 1.day,
+                               equipment_item: model.equipment_items.first)
+            # needed to persist overdue_count
+            model.save
+          end
+
+          it_behaves_like 'calculates availability correctly', 0
         end
-        it { is_expected.to render_template(:edit) }
       end
-      it 'calls delete_files'
     end
-    context 'with non-admin user' do
-      before { sign_in FactoryGirl.create(:user) }
-      it 'should redirect to root' do
-        put :update,
-            id: model,
-            equipment_model: FactoryGirl.attributes_for(:equipment_model)
-        expect(response).to redirect_to(root_url)
+
+    describe 'GET edit' do
```

Comment:
this only tests that permissions and the `before_filter` works; the `#edit` method of the controller is actually empty, so this can probably be deleted